src/core/storage/storage.js

```diff
@@ -24,7 +24,9 @@ export default class Storage {
     /**
      * The listener for window.pop in the persistent storage
      *
-     * @type {Function}
+     * @param {Map<string, string>} queryParamsObject  A Map containing the persisted state
+     * @param {string} queryParamsString the url params of the new state
+     *                                   e.g. "?query=virginia&otherParam=yext"
      */
     this.popListener = (queryParamsObject, queryParamsString) => {
```

Comment:
Going off the JSDoc, I'm not sure what the difference is between these two params. Is `queryParamsObject` just a parsed `queryParamsString`?

Reply:
yeah, that's the only difference. I changed `queryParamsObject` to `queryParamsMap` and gave an example, if that makes the jsdoc clearer?

Reply:
Do we need both pieces of data passed in? It seems like one can be constructed from the other.

Reply:
Yeah it seems a little redundant, we can also get queryParamsString from getUrlWithCurrentState(). This listener's implementation is inside the bespoke answers-storage library, though. Is it alright if we make a separate item to get rid of queryParamsString (or look into getting rid of it?)

Reply:
Yeah, no worries. We can have a separate item.